src/pocketmine/item/Item.php

```diff
@@ -596,7 +596,13 @@ public function addEnchantment(Enchantment $ench){
 		}
 
 		if(!$found){
-			$tag->ench->{count($tag->ench) + 1} = new CompoundTag("", [
+			$count = 0;
+			foreach($tag->ench as $key => $value){
+				if(is_numeric($key)){
+					++$count;
+				}
+			}
+			$tag->ench->{$count + 1} = new CompoundTag("", [
```

Comment:
@Muqsit well, it doesn't care about keys... if 0 1 2 3 4 are set, count is 5, that's basic common sense. I believe you are right though.